### ISSUE TITLE
MPI-2 entry checking

### ISSUE THREAD

**Issue by scottb5612** [SergioMartin86/bamboo6#24]
Bamboo should report any MPI2) entries that it doesn't support 
Any identifier beginning with MPI_
needs to be verified against a list of what  Bamboo currently supports.
I don't know if there is a publicly available list of identifier names,
otherwise you'll need to look at the mpi headers

**Comment by SergioMartin86** [comment]
Probably the best solution is compare every function name starting with MPI_ with the list of functions we support. If it's found, then its ok. If it's not found, then we fail.

This rules out the possibility of using user-created MPI_ named functions, but who is going to do that anyway?

**Comment by SergioMartin86** [comment]
Done just that.

**Comment by scottb5612** [comment]
I gave it a dummy name. We could actually  add our own functions that aren't part of MPI,
like MPI_Glorp, if we supported them!
Ok, one enhancement: rather than quitting upon the first unsupported MPI routine,
Complete the parse for all possible errors
Also, the log output isn't really helpful, since the "not found" errors aren't mentioned there.

Here is a test case

https://www.dropbox.com/s/gjk1flhj8rzxp5i/bad_syn.c?dl=0
